client/core/core.go

```diff
@@ -2161,23 +2236,112 @@ func (c *Core) GetDEXConfig(dexAddr string, certI interface{}) (*Exchange, error
 		return nil, newError(dupeDEXErr, "already registered at %s", dexAddr)
 	}
 
-	dc, err := c.connectDEX(&db.AccountInfo{
+	return c.connectDEX(&db.AccountInfo{
 		Host: host,
 		Cert: cert,
 	}, true)
+}
+
+// GetDEXConfig creates a temporary connection to the specified DEX Server and
+// fetches the full exchange config. The connection is closed after the config
+// is retrieved. An error is returned if user is already registered to the DEX
+// since a DEX connection is already established and the config is accessible
+// via the User or Exchanges methods. A TLS certificate, certI, can be provided
+// as either a string filename, or []byte file contents.
+func (c *Core) GetDEXConfig(dexAddr string, certI interface{}) (*Exchange, error) {
+	dc, err := c.tempDexConnection(dexAddr, certI)
 	if dc != nil {
 		// Stop (re)connect loop, which may be running even if err != nil.
 		defer dc.connMaster.Disconnect()
 	}
 	if err != nil {
-		return nil, codedError(connectionErr, err)
+		return nil, err
 	}
 
 	// Since connectDEX succeeded, we have the server config. exchangeInfo is
-	// guaranteed to return and *Exchange with full asset and market info.
+	// guaranteed to return an *Exchange with full asset and market info.
 	return dc.exchangeInfo(), nil
 }
 
+func (c *Core) PreRegister(dexAddr string, appPW []byte, certI interface{}) (*Exchange, bool, error) {
```

Comment:
Needs some docs to explain the returns (paid mainly), but also when and how PreRegister should be used in conjunction with Register, and when Register can be used on its own.